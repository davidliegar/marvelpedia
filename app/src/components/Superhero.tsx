import type { Superhero } from "@marvelpedia/core";
import Button from "./Button";
import { Icon } from '@iconify/react';

interface SuperheroProps {
  hero: Superhero;
}

function Superhero({ hero }: SuperheroProps) {
  return (
    <article className="bg-white grid grid-rows-[240px_auto_max-content]">
      <img className="object-cover w-full h-full object-center" src={hero.img} alt={hero.name} />
      <div className="p-4">
        <h4 className="text-lg font-semibold py-4">{ hero.name }</h4>
        <p className="text-neutral-600 max-h-28 line-clamp-4">{ hero.description }</p>
      </div>

      <Button className="place-self-end m-4" to={hero.externalLink}>
        <div className="flex items-center gap-2">
          Read more
          <Icon icon="mdi:arrow-right" />
        </div>
      </Button>
    </article>
  )
}

export default Superhero
