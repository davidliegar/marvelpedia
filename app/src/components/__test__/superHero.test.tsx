import { render, screen } from '@testing-library/react';
import Superhero from '../superhero';
import { describe, it, expect } from 'vitest';

// Mock superhero data
const mockHero = {
  id: '1',
  name: 'Spider-Man',
  description: 'description',
  img: 'https://example.com/spiderman.jpg',
  externalLink: 'https://marvel.com/spiderman'
};

describe('Superhero component', () => {
  it('renders the hero image, name, and description', () => {
    render(<Superhero hero={mockHero} />);

    const imgElement = screen.getByAltText('Spider-Man');
    const nameElement = screen.getByText('Spider-Man');
    const descriptionElement = screen.getByText(/description/i);
    const buttonElement = screen.getByRole('link', { name: /read more/i });

    expect(imgElement).toBeInTheDocument();
    expect(imgElement).toHaveAttribute('src', mockHero.img);

    expect(nameElement).toBeInTheDocument();
    expect(descriptionElement).toBeInTheDocument();

    expect(buttonElement).toBeInTheDocument();
    expect(buttonElement).toHaveAttribute('href', mockHero.externalLink);

  });
});
