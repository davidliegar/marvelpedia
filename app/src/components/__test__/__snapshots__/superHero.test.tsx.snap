// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Superhero component > matches the snapshot 1`] = `
<DocumentFragment>
  <article
    class="bg-white grid grid-rows-[240px_auto_max-content]"
  >
    <img
      alt="Spider-Man"
      class="object-cover w-full h-full object-center"
      src="https://example.com/spiderman.jpg"
    />
    <div
      class="p-4"
    >
      <h4
        class="text-lg font-semibold py-4"
      >
        Spider-Man
      </h4>
      <p
        class="text-neutral-600 max-h-28 line-clamp-4"
      >
        description
      </p>
    </div>
    <a
      class="place-self-end m-4 bg-stone-700 hover:bg-stone-900 text-white py-2 px-4 rounded"
      href="https://marvel.com/spiderman"
      rel="noopener noreferrer"
      target="_blank"
    >
      <div
        class="flex items-center gap-2"
      >
        Read more
        <span />
      </div>
    </a>
  </article>
</DocumentFragment>
`;
